helm/botkube/templates/deployment.yaml

```diff
@@ -1,3 +1,5 @@
+{{- $uuid := include "botkube.uuid" . -}}
+
 apiVersion: apps/v1
 kind: Deployment
```

Comment:
Unfortunately it won't work for now - we're missing `automountServiceAccountToken: false` property on Pod: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/

Also, we'll need to adjust the code in `main.go` around

```go
        // Prepare K8s clients and mapper
	kubeConfig, err := clientcmd.BuildConfigFromFlags("", conf.Settings.Kubeconfig)
```
As underneath it uses `InClusterConfig()` which has fixed paths to files. So probably we need to pass them via the env:
![image](https://github.com/kubeshop/botkube/assets/7155799/795b9442-24d2-42d1-8d21-0c00e8802310)

Reply:
I only verified it on resource level as below, I didn't know there is such an effect, on it

<img width="963" alt="Screenshot 2023-05-30 at 14 07 14" src="https://github.com/kubeshop/botkube/assets/1237982/dd0d67c8-9e87-4583-a79b-c20f5a08093c">
<img width="1003" alt="Screenshot 2023-05-30 at 14 07 05" src="https://github.com/kubeshop/botkube/assets/1237982/e47e7745-7d2b-4947-b731-bbfc44f13ae4">

Reply:
Yep, on the resource level is totally fine, but we need to ensure this mounted kubeconfig is used by Botkube Core, and that the automounted token is no longer there so that no plugin will fall back to it 👍 Thanks!

Reply:
@pkosiec I handled the custom sa token path + root ca